### PR TITLE
Fixes #318: enabled prefer-const and semicolon tslint rules

### LINKED ISSUE #318: [lint] enable prefer const tslint rule
https://palantir.github.io/tslint/rules/prefer-const/

And fix existing issues.

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by kittaakos** [comment]
I have disabled the appveyor CI for all branches now. I will enable it as part of #147.

File: packages/tslint.json [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
+      true,
+      "spaces"
+    ],
+    "max-line-length": [
+      true,
+      180
+    ],
     "no-trailing-whitespace": true,
-    "no-unused-expression": true,
-    "no-use-before-declare": true,
```
Comment (marcdumais-work) [inline packages/tslint.json]: I guess we remove this rule because it's mostly applicable to "var", which we disallow in the following rule?

Comment (akosyakov) [inline packages/tslint.json]: I've switched to tslint 5.x and these rules were broken for tslint 4.x. Now we have a lot of warnings for them and mostly invalid like the declaration of interfaces after usages, that is not broken. So I've disabled `no-use-before-declare` and moved `no-unused-expression` to dev.tslint.json.

**Comment by akosyakov** [comment]
I will merge tomorrow if nobody minds.

**Comment by MarkZ3** [comment]
Looks good!

**Comment by hexa00** [comment]
Looks like the yarn.lock contains unrelated updates ?

Otherwise LGTM.

**Comment by akosyakov** [comment]
@hexa00 could be I do `yarn --force`, it updates everything to the latest in the allowed range

**Pull request merged**